lib/asciidoctor/abstract_node.rb

```diff
@@ -221,6 +221,13 @@ def roles
     end
   end
 
+  # Public: A convenience method that adds the given role directly to this node
+  def add_role(name)
+    role_attribute = @attributes['role']
+    roles = role_attribute ? role_attribute.split(' ') : []
+    @attributes['role'] = (roles.dup << name).join(' ')
```

Comment:
I added the `dup` because an extension could create multiple nodes, everytime passing in the *same* value for the attribute *role*.
This is maybe a pathological case and could be handled as a simple "user error" I initially wanted to stay on the safe side.